### PR TITLE
Basic Bridge Server & Integration Tests

### PR DESCRIPTION
**Pull request by indramalav**

### CODE REVIEW

File: src/test/crux-messenger/test-secure-cruxid-messenger-prod-pubsubClientFactory.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                keyManager: new BasicKeyManager(this.user2Data.pvtKey)
+            });
+            user2Messenger.listen((msg) => {
+                resolve(msg)
+            },(err) => {
+                reject(err)
+            });
+            await user1Messenger.send(testmsg, this.user2Data.cruxUser.cruxID);
+        }).then(msg => 
+            expect(msg).equals(testmsg)
```
Comment (quixote911) [inline src/test/crux-messenger/test-secure-cruxid-messenger-prod-pubsubClientFactory.ts]: put this expectation before resolving promise on line 50. no need for this `then` statement

**Review by quixote911 — CHANGES_REQUESTED** [review]

File: src/application/clients/crux-gateway-bridge.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
+import * as net from "net";
```
Comment (quixote911) [inline src/application/clients/crux-gateway-bridge.ts]: Leave this file out, we will add it later when we actually add auth logic

File: src/test/crux-messenger/test-secure-cruxid-messenger-prod-pubsubClientFactory.ts (hunk 1 above)
Comment (indramalav) [reply src/test/crux-messenger/test-secure-cruxid-messenger-prod-pubsubClientFactory.ts]: did it for code symmetry (as in reject scenario) shall I still change it.

Comment (indramalav) [reply src/test/crux-messenger/test-secure-cruxid-messenger-prod-pubsubClientFactory.ts]: @quixote911

File: src/application/clients/crux-gateway-bridge.ts (hunk 2 above)
Comment (indramalav) [reply src/application/clients/crux-gateway-bridge.ts]: changed

File: src/test/crux-messenger/test-secure-cruxid-messenger-prod-pubsubClientFactory.ts (hunk 1 above)
Comment (indramalav) [reply src/test/crux-messenger/test-secure-cruxid-messenger-prod-pubsubClientFactory.ts]: updated this

File: src/core/entities/crux-user.ts [hunk 3]
```diff
@@ -63,6 +63,7 @@ export class CruxUser {
     private cruxUserConfig!: ICruxUserConfiguration;
     private cruxUserPrivateAddresses!: ICruxUserPrivateAddresses;
     private cruxDomain!: CruxDomain;
+    private blacklistedCruxIDs!: string[];
```
Comment (quixote911) [inline src/core/entities/crux-user.ts]: remove all changes in this file. we are not using this feature right now.

File: src/packages/utils.ts [hunk 4]
```diff
@@ -162,6 +166,32 @@ const getKeyPairFromPrivKey = (privKey: string): {
     };
 };
 
+const generateHash = (value: string) => {
```
Comment (quixote911) [inline src/packages/utils.ts]: Remove all unused code

File: src/test/crux-messenger/secure-cruxid-messenger-prod-test.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import * as chai from "chai";
```
Comment (quixote911) [inline src/test/crux-messenger/secure-cruxid-messenger-prod-test.ts]: any tests which interact with the server - put them in `src/test/integration-tests/crux-messenger/`

File: src/test/crux-messenger/secure-cruxid-messenger-prod-test.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+chai.should();
+const expect = require('chai').expect;
+
+
+const user1PvtKey = "KyBuSe1MMV6NjJgZfyWuvgmxUeAYswLC2HrfYLUri9aP3AS5FBfr"; // release020@cruxdev.crux
+const user2PvtKey = "cdf2d276caf0c9c34258ed6ebd0e60e0e8b3d9a7b8a9a717f2e19ed9b37f7c6f"; // mascott6699@cruxdex.crux
+
+describe('Test Secure Crux Messenger - PROD', function() {
+    beforeEach(async function() {
+        const HOST = "127.0.0.1";
```
Comment (quixote911) [inline src/test/crux-messenger/secure-cruxid-messenger-prod-test.ts]: next task is to make server gracefully start up and stop automatiacally in test cases that require it

File: src/application/clients/crux-gateway-bridge-config.json [hunk 7]
```diff
@@ -0,0 +1,21 @@
+{
+    "HOST_URL": {
+        "BROKER_HOST": "127.0.0.1",
+        "TCP_HOST": "127.0.0.1"
+    },
+    "PORTS": {
+        "BROKER_PORT": 1883,
+        "TCP_PORT": 4005,
+        "PUB_SUB_CLIENT_1_PORT": 3000,
```
Comment (quixote911) [inline src/application/clients/crux-gateway-bridge-config.json]: remove every single attribute you are not using in code right now

**Review by quixote911 — CHANGES_REQUESTED** [review]

**Comment by quixote911** [comment]
Github actions is broken - 
https://github.com/cruxprotocol/js-sdk/pull/192/checks?check_run_id=671736027

File: src/application/clients/crux-gateway-bridge-config.json (hunk 7 above)
Comment (indramalav) [reply src/application/clients/crux-gateway-bridge-config.json]: done

File: src/test/crux-messenger/secure-cruxid-messenger-prod-test.ts (hunk 6 above)
Comment (indramalav) [reply src/test/crux-messenger/secure-cruxid-messenger-prod-test.ts]: yes, will figure out and discuss ways to do it

File: src/test/crux-messenger/secure-cruxid-messenger-prod-test.ts (hunk 5 above)
Comment (indramalav) [reply src/test/crux-messenger/secure-cruxid-messenger-prod-test.ts]: done

File: src/packages/utils.ts (hunk 4 above)
Comment (indramalav) [reply src/packages/utils.ts]: done

File: src/core/entities/crux-user.ts (hunk 3 above)
Comment (indramalav) [reply src/core/entities/crux-user.ts]: done and updated related places as well

File: src/test/test-utils.ts [hunk 8]
```diff
@@ -113,7 +113,7 @@ export class InMemoryCruxUserRepository implements ICruxUserRepository {
             }
         },{
             configuration: {
-                blacklistedCruxUsers: [],
+                // blacklistedCruxUsers: [],
```
Comment (quixote911) [inline src/test/test-utils.ts]: Don't comment anything, just remove your changes. Let it be the way it is. You have nothing to do with this file.

File: package.json [hunk 9]
```diff
@@ -14,7 +14,7 @@
     "wallet_demo": "./node_modules/.bin/parcel src/samples/wallet_demo.html --https --no-cache",
     "onboarding_demo": "./node_modules/.bin/parcel src/samples/onboarding_demo.html --https --no-cache",
     "typecheck": "./node_modules/.bin/tsc --noEmit",
-    "validate_lockfile": "./node_modules/.bin/lockfile-lint --type npm --path package-lock.json --validate-https --allowed-hosts npm",
+    "validate_lockfile": "./node_modules/.bin/lockfile-lint --type npm --path package-lock.json --allowed-hosts npm github.com --allowed-schemes \"https:\" \"git+https:\"",
```
Comment (quixote911) [inline package.json:20]: what is this change for?

File: src/application/clients/crux-wallet-client.ts [hunk 10]
```diff
@@ -230,27 +230,27 @@ export class CruxWalletClient {
...
-                registeredCruxUsers.push(fullCruxID);
-            }
-        }
-        cruxUserWithKey.setBlacklistedCruxIDs(registeredCruxUsers);
-        await this.cruxUserRepository.save(cruxUserWithKey, this.getKeyManager());
-        return {success: registeredCruxUsers, failures};
-    }
+    // @throwCruxClientError
+    // public blacklistUsers = async (fullCruxIDs: string[]): Promise<{success: string[], failures: string[]}> => {
+    //     await this.initPromise;
```
Comment (quixote911) [inline src/application/clients/crux-wallet-client.ts]: don't comment anything, just don't change anything that's present. your changes do not intersect.
earlier I had not merged sanchay's `blacklistUsers` feature branch

File: src/core/entities/crux-user.ts [hunk 11]
```diff
@@ -63,6 +63,7 @@ export class CruxUser {
     private cruxUserConfig!: ICruxUserConfiguration;
     private cruxUserPrivateAddresses!: ICruxUserPrivateAddresses;
     private cruxDomain!: CruxDomain;
+    // private blacklistedCruxIDs!: string[];
```
Comment (quixote911) [inline src/core/entities/crux-user.ts]: remove changes

File: src/core/entities/crux-user.ts [hunk 12]
```diff
@@ -92,6 +93,9 @@ export class CruxUser {
     get privateAddresses() {
         return this.cruxUserPrivateAddresses;
     }
+    // get blacklist() {
+    //     return this.blacklistedCruxIDs;
```
Comment (quixote911) [inline src/core/entities/crux-user.ts]: remove changes

File: package.json (hunk 9 above)
Comment (indramalav) [reply package.json:20]: @SanchayPahalwani can you explain please

File: src/application/clients/crux-wallet-client.ts (hunk 10 above)
Comment (indramalav) [reply src/application/clients/crux-wallet-client.ts]: I am removing all blacklist wala code

File: src/core/entities/crux-user.ts (hunk 11 above)
Comment (indramalav) [reply src/core/entities/crux-user.ts]: will do

Comment (malavindra5) [reply src/core/entities/crux-user.ts:66]: will do

File: src/core/entities/crux-user.ts (hunk 12 above)
Comment (indramalav) [reply src/core/entities/crux-user.ts]: will do it

File: src/test/test-utils.ts (hunk 8 above)
Comment (indramalav) [reply src/test/test-utils.ts]: done

File: src/application/clients/crux-wallet-client.ts (hunk 10 above)
Comment (indramalav) [reply src/application/clients/crux-wallet-client.ts]: done

File: src/core/entities/crux-user.ts (hunk 11 above)
Comment (indramalav) [reply src/core/entities/crux-user.ts]: done

File: src/core/entities/crux-user.ts (hunk 12 above)
Comment (indramalav) [reply src/core/entities/crux-user.ts]: done

File: package.json (hunk 9 above)
Comment (quixote911) [reply package.json:20]: @SanchayPahalwani

**Review by quixote911 — APPROVED** [review]

File: package.json (hunk 9 above)
Comment (SanchayPahalwani) [reply package.json:20]: This package is to verify all npm dependencies are coming from npm in the package-lock.json and wasn't passing the check for one of the strong pubsub package that's why github was also added as a valid host here

**Pull request merged**